lib/helpers/timings.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+var NS_PER_SEC = 1e9;
+var MS_PER_NS = 1e6;
+
+/**
+* Get duration in milliseconds from two process.hrtime()
+* @function getHrTimeDurationInMs
+* @param {Array} startTime - [seconds, nanoseconds]
+* @param {Array} endTime - [seconds, nanoseconds]
+* @returns {Number|null} durationInMs
+*/
+function getHrTimeDurationInMs (startTime, endTime) {
+    if (startTime === null || endTime === null) {
```

Comment:
When would these be null? In the case of incomplete/aborted requests that cause only parts of the timing to be recorded?

Reply:
When for example url is an IP so there is no DNS resolve phase, or no TLS etc.
It's not related to the request start/end time, it's more like a from-to for two htimes, maybe the naming is misleading.